solr/core/src/java/org/apache/solr/core/ZkContainer.java

```diff
@@ -112,20 +111,14 @@ public void initZooKeeper(final CoreContainer cc, CloudConfig config) {
           throw new ZooKeeperException(SolrException.ErrorCode.SERVER_ERROR,
               "A chroot was specified in ZkHost but the znode doesn't exist. " + zookeeperHost);
         }
-        ZkController zkController = new ZkController(cc, zookeeperHost, zkClientConnectTimeout, config,
-            new CurrentCoreDescriptorProvider() {
-
-              @Override
-              public List<CoreDescriptor> getCurrentDescriptors() {
-                List<CoreDescriptor> descriptors = new ArrayList<>(
-                    cc.getLoadedCoreNames().size());
-                Collection<SolrCore> cores = cc.getCores();
-                for (SolrCore core : cores) {
-                  descriptors.add(core.getCoreDescriptor());
-                }
-                return descriptors;
-              }
-            });
+        ZkController zkController = new ZkController(cc, zookeeperHost, zkClientConnectTimeout, config, () -> {
```

Comment:
I'd prefer that you declare that last parameter as a local typed variable and then pass it in.  This makes it clearer what it is that is being passed to ZkController's c'tor.